### ISSUE TITLE
Bug in elementIsTreeNode function returning incorrect 'outofbounds' variable

### ISSUE THREAD

**Issue by davechisholm** [angular-ui-tree/angular-ui-tree#763]
My nodes have the structure

```
<li ui-tree-node>
  <div ui-tree-handle></div>
  <div class="content">node content here</div>
  <ul ui-tree-nodes>...</ul>
</li>
```

While dragging, the 'outOfBounds' variable is returning true whenever I drag over my content element. This is because the elementIsTreeNode function is checking if the target element has the attribute ui-tree-node. In this case, it is the _parent_ of the target node that has the ui-tree-node attr. The result is jerky / randomly flashing dragging. I have amended the outofBounds variable as below which appears to have fixed the issue.

```
                outOfBounds = !UiTreeHelper.elementIsTreeNodeHandle(targetElm) &&
                              !UiTreeHelper.elementIsTreeNode(targetElm) &&
                              !UiTreeHelper.elementIsTreeNodeChild(targetElm) && //added
                              !UiTreeHelper.elementIsTreeNodes(targetElm) &&
                              !UiTreeHelper.elementIsTree(targetElm) &&
                              !UiTreeHelper.elementIsPlaceholder(targetElm);

  //added below the elementIsTreeNode function
          elementIsTreeNodeChild: function (element) {
            return typeof angular.element(element[0].parentElement).attr('ui-tree-node') !== 'undefined';
          },
```

**Comment by dturing** [comment]
+1; i also have a child within my ui-tree-node and dropping becomes a game of chance, wether you're hitting that child or not.

@davechisholm's suggestion of adding elementIsTreeNodeChild works perfectly for me.

**Comment by rribou** [comment]
Same problem here, but davechisholm's solution seems to fix it.

**Comment by Voles** [comment]
#856 should fix this. Released in v2.22.2
